service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/CodeableConceptCdMapper.java

```diff
@@ -87,6 +87,43 @@ public String mapCodeableConceptToCd(CodeableConcept codeableConcept) {
         return TemplateUtils.fillTemplate(CODEABLE_CONCEPT_CD_TEMPLATE, builder.build());
     }
 
+    // Medications are currently using D&T Codes rather than snomed codes but are being passed through as SNOMED codes which is
+    // creating a degradation on the receiving side. Until the types are configured correctly and agreed to a specification
+    // we have agreed to use the Concept ID rather than Description Id for medications which will avoided the degradation.
+    public String mapCodeableConceptForMedication(CodeableConcept codeableConcept) {
+        var builder = CodeableConceptCdTemplateParameters.builder();
+        var mainCode = findMainCode(codeableConcept);
+
+        builder.nullFlavor(mainCode.isEmpty());
+
+        if (mainCode.isPresent()) {
+            var extension = retrieveDescriptionExtension(mainCode.get())
+                .map(Extension::getExtension)
+                .orElse(Collections.emptyList());
+
+            builder.mainCodeSystem(SNOMED_SYSTEM_CODE);
+
+            Optional<String> code = Optional.of(mainCode.get().getCode().toString());
```

Comment:
Is it possible for `getCode` to be null here. If so should we use `Optional.ofNullable()` in case there is a `NullPointerException`?

Reply:
Good Shouts, there seems to be a pattern in this class where potential nullables are being ignored. I've now been through the class and corrected all instances.